pkg/preparer/listener.go

```diff
@@ -79,13 +79,23 @@ func (l *HookListener) Sync(quit <-chan struct{}, errCh chan<- error) {
 			}
 
 			hookPod := pods.NewPod(result.Manifest.ID(), path.Join(l.DestinationDir, event, result.Manifest.ID()))
-			// install hooks as current user, not the pod ID. TODO: authorization checks for the pod
-			// to ensure they are permitted.
-			curUser, err := user.Current()
+
+			// Hooks may specify a user to run as - by default they will run as their pod ID.
+			topLevelConfig := config.LoadFromUnpacked(result.Manifest.Config)
+			hookConfig, err := topLevelConfig.ReadMap("hook")
+			if err != nil {
+				sub.WithField("err", err).Errorln("Hook configuration is invalid")
+				break
+			}
+			runAs, err := hookConfig.ReadString("run_as")
```

Comment:
We had talked about a dot syntax for that at one point

`config.ReadString("foo.bar.baz")`

May come back to that in the future